Show item data when hovering the mouse over the Item

GreenJon902 opened the pull request:
When you hover your mouse over an item when inspecting a chest, it will show the custom name of that item and also the customModelData (Only if set to true in config).
![image](https://user-images.githubusercontent.com/37347905/168463485-92bc1ffa-d625-4d1d-80a3-b9e325ab6e80.png)
![image](https://user-images.githubusercontent.com/37347905/168463535-62a835ce-0deb-4a96-95e3-193d7ed9a1c6.png)
![image](https://user-images.githubusercontent.com/37347905/168463546-8685aa69-bcc9-425c-8de0-02d54738e248.png)
If an item does not have any customModelData or does not have a custom name then it will not be shown, if they have neither then no hover dialogue opens!
I did have to modify [PlayerInteractListener.java](https://github.com/GreenJon902/CoreProtect/blob/2242ba8f45ea1c41a0c4125a2c0c73484547bfaf/src/main/java/net/coreprotect/listener/player/PlayerInteractListener.java#L310) to add newlines that don't get sent as their own messages by replacing `"\\n"` with `"\n"` after the string with `"\n"` was split up.

vacla commented on src/main/java/net/coreprotect/database/lookup/ChestTransactionLookup.java:
```diff
@@ -1,20 +1,27 @@
 package net.coreprotect.database.lookup;
 
-import java.sql.ResultSet;
```
Please revert the formatting to how it was previously

vacla commented on src/main/java/net/coreprotect/database/lookup/ChestTransactionLookup.java:
```diff
@@ -117,6 +125,29 @@ else if (commandSender.hasPermission("coreprotect.coreprotect")) {
+                    }
+                }
+
+                //                                                                                                                                          Keep these in case no hover and target is default    \/              \/
```
Change to a normal comment like on line 123

vacla commented:
@GreenJon902 would it be possible to also add this metadata info to the networking?

GreenJon902 commented:
@vacla I believe I have fixed the things you asked me to. By networking did you mean PluginChannelListener as I have added the metadata to that. I do not understand what it is or how to test it but it does not throw any errors.

vacla commented:
I'm currently making some improvements for future cases

vacla commented:
@GreenJon902 since i can't directly make changes to this PR, I made a PR to your branch: https://github.com/GreenJon902/CoreProtect/pull/1

GreenJon902 commented:
@vacla I have merged your pull request, thank you for the help!

vacla commented:
No worries, I'm a developer myself that implemented the whole Networking and I will use it for my mod called Watson so now it will be able to show that info also.

vacla commented on src/main/java/net/coreprotect/listener/channel/PluginChannelListener.java:
```diff
@@ -66,6 +81,15 @@ public void sendData(CommandSender commandSender, long timeAgo, Phrase phrase, S
+            {
+                for (Map.Entry<String, String> additionalItem : additional.entrySet())
+                {
+                    msgOut.writeUTF(additionalItem.getKey());
```
Woops, i made a fault here, can you change the writeUTF to Chat.console please?

GreenJon902 replied:
Both of them or just the first one?

vacla replied:
Both at that position since it will already be added above, in the if statement for network debug is just to sent to console.

GreenJon902 replied:
Done!

vacla commented on src/main/java/net/coreprotect/database/lookup/ChestTransactionLookup.java:
```diff
@@ -117,8 +130,49 @@ else if (commandSender.hasPermission("coreprotect.coreprotect")) {
+                            }
+
+                            popupText += Color.WHITE + "\\n - " + Color.DARK_AQUA + name + " " + Color.GREY + itemMeta.getEnchantLevel(enchant);
+                            additionalData.put("enchantments", additionalData.get("enchantments") + name + "-" + itemMeta.getEnchantLevel(enchant) + "|");
```
i assume this would look like `sharpness-2|mending-1|` ? If so, would it be possible to remove the last `|` at the end?

GreenJon902 commented:
The co lookup command can now see the item enchants, custom names, and custom model data (if enabled)!

JHarris12345 commented:
Was this added into main core protect? How can I get the version with it in?

GreenJon902 commented:
No, I believe it's still waiting to merged into the main release. If you really want this feature then you could probably compile it yourself from my fork of this repository (https://github.com/GreenJon902/CoreProtect).

Intelli commented:
Tooltips for item names and enchantments have been implemented here: https://github.com/PlayPro/CoreProtect/commit/5d6fd60d40281f51fbd248dc7099e008de67b862
